pypeit/setup_gui/model.py

```diff
@@ -0,0 +1,1469 @@
+            if orientation == Qt.Orientation.Horizontal and section < len(self.colnames):
+                # Columns have propper names
+                return self.colnames[section]
+            else:
+                return " "
+        else:
+            # A non-applicable role or a sort order request for a column that we're not sorted by.
+            return None
+
+    def getSetup(self, row : int) -> dict:
+        """Return the configuration/setup values for the given row of metadata.
+        
+        Args:
+            row: The row within the model to return the setup values for
+        Return:
+            A dictionary of configuration key/value pairs. An empty dictionary is returned
+            if there is no metadata or the spectrograph has no configuration keys
+        """
+        if self.metadata is None or row >= len(self.metadata):
+            return {}
+        return self.metadata.get_configuration(row)
+
+
+
+    def getDefaultColumns(self):
+        """Return the default columns to display to the user. This can vary based
+        on the current spectrograph.
+        
+        Return:
+        A list of column names, in order to display.
+        """
+        if self.metadata is not None:
+            default_columns = self.metadata.set_pypeit_cols(write_bkg_pairs=True)
+            return default_columns
+        else:
+            return ['filename', 'frametype', 'ra', 'dec', 'target', 'dispname', 'decker', 'binning', 'mjd', 'airmass', 'exptime']
+
+    def getStringColumnSize(self, colname: str) -> int | None:
+        """
+        Return the maximum size of a string column.
+
+        Args:
+            colname: The name of the column
+        Return:
+            The maximum size of the column, or None if it is not a string column.
+        """
+        dt = self.metadata[colname].dtype
+        if dt.kind == "U":
+            # Divide the size by the size for a single character column
+            return int(dt.itemsize / self._dtype_char_size)
+        else:
+            # Not a string
+            return None
+
+    def resizeStringColumn(self, colname : str, new_size : int)->None:
+        """Resize a string column to fit a new size.
+        
+        Args:
+            colname: The name of the string column.
+        """
+        # Create a new column with the new size.
+        # We do this before setting the column, so that an exception will leave the
+        # metadata in the previous state
+        new_column = astropy.table.Column(self.metadata[colname],dtype=f'<U{new_size}')
+
+        self.metadata[colname] = new_column
+
+    def reset(self):
+        """
+        Reset the proxy assuming the metadata object has completely changed
+        """
+        # Tell views a model reset is happening
+        super().beginResetModel()
+
+        # Reset column names
+        self.colnames = self.getDefaultColumns()
+
+        super().endResetModel()
```

Comment:
Here and above you can use `self` instead of `super` because this subclass doesn't override the base class functions, right?